pkg/apis/samples/v1/types.go

```diff
@@ -349,8 +349,9 @@ func (s *Config) ClusterOperatorStatusAvailableCondition() (configv1.ConditionSt
 // 2) the first string is the succinct text to apply to the Progressing condition on failure
 // 3) the second string is the fully detailed text to apply the the Failing condition
 func (s *Config) ClusterOperatorStatusFailingCondition() (configv1.ConditionStatus, string, string) {
-	if len(s.Status.Conditions) == 0 {
-		// first event, have not processed default config yet
+	// do not start checking for bad config and needed cred until we've iterated through
+	// the credential / config processing to actually processed a config
+	if len(s.Status.Conditions) < 3 {
```

Comment:
FYI I had meant to rework this a bit, such that it confirms the init status conditions for all the conditions has occurred as part of the first config pass

I'll push that tommorrow